src/network.c

```diff
@@ -2684,6 +2717,43 @@ static int network_config_set_interface(const oconfig_item_t *ci, /* {{{ */
   return 0;
 } /* }}} int network_config_set_interface */
 
+static int network_config_set_bind_address(const oconfig_item_t *ci,
+                                        struct sockaddr_storage **bind_address) {
+  char addr_text[256];
+
+  if (cf_util_get_string_buffer(ci, addr_text, sizeof(addr_text)) != 0)
+    return -1;
+
+  int ret;
+  struct addrinfo hint, *res = NULL;
+
+  memset(&hint, '\0', sizeof hint);
+  hint.ai_family = PF_UNSPEC;
+  hint.ai_flags = AI_NUMERICHOST;
+
+  ret = getaddrinfo(addr_text, NULL, &hint, &res);
+  if (ret) {
+    ERROR("Invalid address");
+    ERROR(gai_strerror(ret));
+    return 1;
+  }
+
+  *bind_address = malloc(sizeof(**bind_address));
```

Comment:
Someone can try to put several `BindAddress` options into config.
I think we need take this into account.

Reply:
Allocated memory should be freed at some moment.
Please take a look into `sockent_destroy() / free_sockent_client()`.